### DISCUSSION TITLE
Check for local file updates

### DISCUSSION THREAD

**Discussion by awesomescot** [copier-org/copier#1467] [Q&A]
Hi.  I'm trying to write a script for our CI system that will check to see if any templated files have been modified.  In our repo we have some files that we are trying to template, but I'm sure other developers will try to modify them directly.  I want to have a test that will fail if any files have been modified from their templated form.  We are using a directory as the template source, so we can't use any git fanciness for this.

Basically I just want to run `copier copy` and instead of asking to update files that are different, just exit with a nonzero exit code.

Thanks!

**Comment by pawamoy** [comment]
Generate a temporary project elsewhere and compare the files that should not change?

**Reply by awesomescot** [reply]
That would work.  Should I add a feature request to add a --check flag?  That would be super nice to have.

**Reply by pawamoy** [reply]
Isn't that easy enough with these two commands?

```
copier copy template /tmp/project
diff -qrX file_containing_patterns_to_exclude /tmp/project .
```

That's what I was suggesting :)

**Reply by awesomescot** [reply]
Well it's never that simple.  Here's what I ended up with.
```
copier copy --data-file $DEST_DIR/.copier-answers.yml $TEMPLATE_DIR $TEMP_DIR

# check if there are any changes between backup and updated files
filelist=( "filename1" "filename2" )
for filename in "${filelist[@]}"; do
  if ! diff $TEMP_DIR/$filename $DEST_DIR/$filename; then
    echo "ERROR: Templated file (${filename}) has been modified locally without updating template."
    diff -u $TEMP_DIR/$filename $DEST_DIR/$filename
  fi
done
```
Which isn't horrible.  But it could have been as simple as this.
```
if ! copier copy --check --defaults $TEMPLATE_DIR $DEST_DIR; then
   echo "ERROR..."
fi
```
Which seems easier, but I totally understand if there are a million other priorities.
Thanks for the help!
